docs/extensions/bastion.md

```diff
@@ -0,0 +1,42 @@
+# Contract: `Bastion` resource
+
+The Gardener project allows users to connect to Shoot worker nodes via SSH. As nodes are usually firewalled and not directly accessible from the public internet, [GEP-15](../proposals/15-manage-bastions-and-ssh-key-pair-rotation.md) introduced the concept of "Bastions". A bastion is a dedicated server that only serves to allow SSH ingress to the worker nodes.
+
+`Bastion` resources are usually created by `gardenctl` when a user wants to establish an SSH connection. They contain the user's public SSH key and IP address, in order to provision the server accordingly: The public key is put onto the Bastion and SSH ingress is only authorized for the given IP address (in fact, it's not a single IP address, but a set of IP ranges, however for most purposes a single IP is be used).
```

Comment:
As of the latest update of the GEP, it's not the user's public SSH key but a randomly computed key. Anyways, that's an implementation detail of `gardenctl`, so for this document I would just write

> contain a public SSH key

Reply:
Generally, as this doc just describes the "API"/extension contract from g/g's point of view, maybe let's not talk about `gardenctl` at all here?

Reply:
I am unsure where this documentation should end and the proposal begins. I didn't want to repeat things from the proposal, otherwise 2 places might go out of sync with reality over time. ^^

Reply:
How is this related to the suggestion not talking about one special user of this API (`gardenctl`) but only about the API in general? 👀